### PR TITLE
cmd/compact: add new compact cmd, used for specific path

### LINKED ISSUE #4303: Support triggering compaction of all files under a given directory
**What would you like to be added**:
Support triggering compaction of all files under a given directory

**Why is this needed**:
An alternative light way of `gc --compact` for large clusters.

### PR DESCRIPTION
**Pull request by jiefenghuang**
close #4303

### CODE REVIEW

File: cmd/compact_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+/*
+ * JuiceFS, Copyright 2020 Juicedata, Inc.
```
Comment (davies) [inline cmd/compact_test.go:2]: 2024

File: pkg/utils/file.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+package utils
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+func GetInode(path string) (uint64, error) {
+	dpath, err := filepath.Abs(path)
```
Comment (davies) [inline pkg/utils/file.go:25]: Is this necessary?

File: pkg/meta/sql_lock.go [hunk 3]
```diff
@@ -249,9 +249,9 @@ func (m *dbMeta) Setlk(ctx Context, inode Ino, owner_ uint64, block bool, ltype
 	return err
 }
 
-func (r *dbMeta) ListLocks(ctx context.Context, inode Ino) ([]PLockItem, []FLockItem, error) {
+func (m *dbMeta) ListLocks(ctx context.Context, inode Ino) ([]PLockItem, []FLockItem, error) {
```
Comment (davies) [inline pkg/meta/sql_lock.go:252]: We can move these into another PR.

File: cmd/compact_unix.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	compactConcurrency := ctx.Int("compact-concurrency")
+	if compactConcurrency <= 0 {
+		logger.Warn("thread number should be > 0")
+		compactConcurrency = 1
+	}
+
+	// new meta client
+	metaConf := meta.DefaultConf()
+	metaConf.MaxDeletes = int(deleteConcurrency)
+	client := meta.NewClient(metaUri, metaConf)
```
Comment (SandyXSD) [inline cmd/compact_unix.go:105]: Don't use meta client, instead write control message to a mount point. Check these codes:

- https://github.com/juicedata/juicefs/blob/7b85645cac16c8442b3b7d35365a99dc0e339cdd/cmd/info.go#L116
- https://github.com/juicedata/juicefs/blob/7b85645cac16c8442b3b7d35365a99dc0e339cdd/pkg/vfs/internal.go#L297
